### PR TITLE
PD-1203 Updates to Storj App Content

### PR DESCRIPTION
**Pull request by micjohnson777**
This PR updates the existing Storj app tutorial, brining it up-to-date for the 24.04 and upcoming 24.10 releases.

It updates the port assignments, and app deployment instructions and screenshots.

It changes list types in the Before you Begin and Installing Storj app sections.

It adds a note about the old port number assignment because there is enough content out in Google searches about port 20988 instead of 28967 that this content should clear up any confusion on that point. 

It also adds information about Storj multi-tenant solution architecture not being included. A user became confused about this requesting information on a datastore that is not covered in our article. This is not something the SCALE Storj app can be configured for as it is provided by Storj  to customers on the Storj site, not through the node.

Thanks for contributing to TrueNAS documentation! By opening a Pull Request, you're acknowledging that your changes will be distributed under the [Creative Commons 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license.

### CODE REVIEW

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 1]
```diff
@@ -15,151 +15,215 @@ keywords:
...
 
-   Alternatively, use a dynamic DNS (DDNS) service such as NoIP to [create a host name](#creating-a-ddns-host-name) if you do not have a static IP address for the system nodes.
+  Every node must have a unique identifier on the network. Use NFS/SMB shares or a file transfer service such as FTP to upload the credentials generated.
+  If the identity is not present in the storage directory, it generates and authorizes one automatically.
+  This can take a long time and consume system resources while it generates one.
 
-6. Create a publicly-available domain name to access the Storj application. Point this to your router public IP address.
+* Prepare TrueNAS SCALE for the app installation by [creating the datasets](#creating-the-storj-datasets-on-truenas-scale) for the application storage, and a new user if not using the default apps user.
+   
+   Create a parent dataset, such as *storj-node* and then the two storage datasets (**identity** and **config**) under it.
```
Comment (stavros-k) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:54]:
```suggestion
   Create a parent dataset, such as *storj-node* and then the two storage datasets (**identity** and **data**) under it.
```
lets use data to match the UI label

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 2]
```diff
@@ -15,151 +15,215 @@ keywords:
...
+   Select **apps** as the **Dataset Preset** for these datasets. You can modify the dataset ACLs at the time of creation, or modify them later when adding them in the app.
 
-7. [Create a Storj identity and authorize it](https://docs.storj.io/node/dependencies/identity) for every node.
-   Every node must have a unique identifier on the network. Use NFS/SMB shares or a file transfer service such as FTP to upload the credentials generated.
-   If the identity is not present in the storage directory, it generates and authorizes one automatically.
-   This can take a long time and consume system resources while it generates one.
+   If not using the default user **apps** (568) as the assigned user and group, go to **Credentials > Local User**, click **Add** and create the new user.
+   Make sure the user has read/write/execute privileges. If you have an existing admin user on your system with the right permissions, you make that user the app user.
```
Comment (stavros-k) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:58]: execute shouldn't be needed for **data **

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 3]
```diff
@@ -15,151 +15,215 @@ keywords:
...
+Alternatively, use a dynamic DNS (DDNS) service such as NoIP to [create a host name](#creating-a-ddns-host-name) if you do not have a static IP address for the system nodes.
 
+## Creating a DDNS Host Name
 Create a DDNS host name that points to your router WAN IP address, and provide a domain name to use for accessing the Storj application.
-You can use a dynamic DNS service that allows you to set up a DDNS host name. You can use a service such as NoIP to create a domain name (i.e., *name.ddns.net*) and then point it at the WAN IP address of your router.
+You can use a dynamic DNS service that allows you to set up a DDNS host name.
+You can use a service such as NoIP to create a domain name (i.e., *name.ddns.net*) and then point it at the WAN IP address of your router.
 
-Use <code>nislookup <i>name.ddns.net</i></code> to verify it works.
+To verify it works, use <code>nislookup <i>name.ddns.net</i></code> where <code><i>name.ddns.net</i></code> is your DDNS host name.
```
Comment (stavros-k) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:111]:
```suggestion
To verify it works, use <code>nslookup <i>name.ddns.net</i></code> where <code><i>name.ddns.net</i></code> is your DDNS host name.
```

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 4]
```diff
@@ -15,151 +15,215 @@ keywords:
...
+Log into TrueNAS SCALE, then go to **Datasets** and click **Add Dataset** to open the **Add Dataset** screen.
 
-{{< trueimage src="/images/SCALE/Datasets/AddDatasetStorjSCALE.png" alt="Add Dataset Storj SCALE" id="Add Dataset Storj SCALE" >}}
+{{< trueimage src="/images/SCALE/Datasets/AddDatasetNameAndOptions.png" alt="Name and Options" id="Name and Options" >}}
 
-2. Enter a name for the first dataset in **Name**. For example, *storj-node*, and click **Save**.
+1. Enter a name for the first dataset in **Name**. For example, *storj-node*, select **Apps** in **Dataset Preset**, then click **Save**.
 
-3. Select the new dataset *storj-node*, click **Add Dataset** again to create a new child dataset. For example, *config*.
+2. Select the new dataset *storj-node*, click **Add Dataset** again to create a new child dataset. For example, **config**.
```
Comment (stavros-k) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:122]:
```suggestion
2. Select the new dataset *storj-node*, click **Add Dataset** again to create a new child dataset. For example, **data**.
```

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 5]
```diff
@@ -15,151 +15,215 @@ keywords:
...
 
-4. Click **Save**.
+3. Select the *storj-node* dataset again, click **Add Dataset** and create the second child dataset. For example, **identity**.
+   Select **Apps** in **Dataset Preset**, then click **Save**.
 
-5. Select the *storj-node* dataset again, click **Add Dataset** and create the second child dataset. For example, *identity*.
+   {{< trueimage src="/images/SCALE/Datasets/DatasetsDashboardStorjNodeNested.png" alt="Nested Datasets for Storj" id="Nested Datasets for Storj" >}}
 
-6. Click **Save**.
+The **Datasets** screen displays two nested datasets **config** and **identity** underneath the **storj-node** dataset.
```
Comment (stavros-k) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:130]:
```suggestion
The **Datasets** screen displays two nested datasets **data** and **identity** underneath the **storj-node** dataset.
```

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 6]
```diff
@@ -15,151 +15,215 @@ keywords:
...
+   {{< trueimage src="/images/SCALE/Apps/InstallStorjNetworkConfig.png" alt="Storj Network Settings" id="Storj Network Settings" >}}
 
-{{< trueimage src="/images/SCALE/Apps/InstallStorjAppNetworkingSCALE.png" alt="Network Settings Storj App SCALE" id="Network Settings Storj App SCALE" >}}
+4. Accept the default **Network Configuration** settings. 
+   If you want to use different ports, check the [Default Ports](https://www.truenas.com/docs/references/defaultports/) for a list of available or unavailable port numbers.
+   Leave **Host Network** unselected.
 
-9. Review your entries and then click **Save**.
+5. Configure the **Storage Configuration** settings.
+   You can use the default **ixVolume** option to allow SCALE to create the data storage volume or select **Host Path** to select the path to the new **config** dataset created earlier in this document.
```
Comment (stavros-k) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:194]:
```suggestion
   You can use the default **ixVolume** option to allow SCALE to create the data storage volume or select **Host Path** to select the path to the new **data** dataset created earlier in this document.
```

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md (hunk 2 above)
Comment (micjohnson777) [reply content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:58]: I'll add the note about the **data** dataset permissions in the section on the ACL. This part is talking about the user permissions in general.

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 7]
```diff
@@ -15,151 +15,219 @@ keywords:
...
 
-1. Review the Storj node hardware and bandwidth considerations at [Storj Node](https://www.storj.io/node).
+* Review the Storj node hardware and bandwidth considerations at [Storj Node](https://www.storj.io/node).
+
+* Update [TrueNAS SCALE]({{< relref "UpdateSCALE.md" >}}) to the latest public release.
+
+* Create a [wallet address](#getting-a-wallet-address).
 
-2. Update [TrueNAS SCALE]({{< relref "UpdateSCALE.md" >}}) to the latest public release.
+* Obtain a [Storj authentication token](#generating-an-authentication-token-for-storj).
```
Comment (DjP-iX) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:40]:
```suggestion
* Obtain a [Storj authentication token](#obtaining-an-authentication-token-for-storj).
```

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md (hunk 2 above)
Comment (DjP-iX) [reply content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:58]:
```suggestion
   Make sure the user has read/write/execute privileges. If you have an existing admin user on your system with the right permissions, you can make that user the app user.
```

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 8]
```diff
@@ -15,151 +15,219 @@ keywords:
...
-   If the identity is not present in the storage directory, it generates and authorizes one automatically.
-   This can take a long time and consume system resources while it generates one.
+   If not using the default user **apps** (568) as the assigned user and group, go to **Credentials > Local User**, click **Add** and create the new user.
+   Make sure the user has read/write/execute privileges. If you have an existing admin user on your system with the right permissions, you make that user the app user.
 
-8. [Install the Storj application](#installing-the-storj-app) in SCALE.
+   If using ports other than the assigned default ports (**20909** and **28967**), go to [Default Ports](https://www.truenas.com/docs/references/defaultports/) for a list of available or unavailable port numbers to verify the port numbers you want to use are available options.
+
+   {{< hint type=info >}}
+   Earlier releases of SCALE used port 20988 for in and outbound traffic. This port assignment changed in later releases to 28967.
```
Comment (DjP-iX) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:63]: Was this specifically Storj node traffic? Apps traffic? More general?

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 9]
```diff
@@ -15,151 +15,219 @@ keywords:
...
+   {{< /hint >}}
+
+* [Install the Storj application](#installing-the-storj-app) in SCALE.
 
 Storj provides a [Quickstart Node Setup Guide](https://docs.storj.io/node/setup) with step-by-step instructions to help users create a Storj node.
 
 ## Getting a Wallet Address
+Create a wallet address using any crypto wallet provider such as kraken.com, coinbase.com, Bitcoin Wallet, etc.
+If you already have a wallet for other crypto-currency application, you can use the existing wallet.
```
Comment (DjP-iX) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:72]: either
```suggestion
If you already have a wallet for another crypto-currency application, you can use the existing wallet.
```
or
```suggestion
If you already have a wallet for other crypto-currency applications, you can use the existing wallet.
```

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 10]
```diff
@@ -15,151 +15,219 @@ keywords:
...
+Do not change the **Version**. SCALE alerts you when a new version is available for deployed apps, and allows you to update through the click of a button on the **Installed** application screen.
 
-3. Enter the [authentication token copied from Storj](#generating-an-authentication-token-for-storj) in **Configure Auth token for Storj Node**.
-   Enter the email address associated with the token in **Configure Email for Storj**.
+1. Enter the **Storj Configuration** settings:
 
-4. Enter the [storage domain](#creating-a-ddns-host-name) (i.e., the public network DNS name) added for Storj in **Add Your Storage Domain for Storj**.
+   a. [Enter the wallet address](#getting-a-wallet-address) in **Wallet** in the **Storj Configuration** section.
+
+   {{< trueimage src="/images/SCALE/Apps/InstallStojConfigurationScreen.png" alt="Storj Configuration Settings" id="Storj Configuration Setting" >}}
```
Comment (DjP-iX) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:164]: There's a typo in this filename also, but it's working so we can probably ignore it
```suggestion
   {{< trueimage src="/images/SCALE/Apps/InstallStojConfigurationScreen.png" alt="Storj Configuration Settings" id="Storj Configuration Settings" >}}
```

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md [hunk 11]
```diff
@@ -15,151 +15,219 @@ keywords:
...
+   If you add an ACE (ACL entry) for this dataset, you do not need to grant execute permissions for the user added for this dataset.
+   {{< /hint >}}
 
-Environmental variables are optional.
-If you want to include additional variables, see [Storj Environment Variables](https://storj.github.io/core/tutorial-environment-variables.html) for a list.
-Click **Add** for each variable you want to add.
+   Repeat this for the **identity** dataset. Select **Host Path** in the **Storj Identity Storage > Type** field then enter or browse to select the mount path for the existing dataset.
+
+   If you created additional datasets for other purposes and want to add that additional storage, click **Add** to the right of **Additional Storage Volumes** to show another set of storage options.
+   The additional storage options include the **ixVolume** and **Host Path** options, and the otpion to create an **SMB share** volume.
```
Comment (DjP-iX) [inline content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:207]:
```suggestion
   The additional storage options include the **ixVolume** and **Host Path** options, and the option to create an **SMB share** volume.
```

**Review by DjP-iX — CHANGES_REQUESTED** [review]

File: content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md (hunk 8 above)
Comment (micjohnson777) [reply content/SCALE/SCALETutorials/Apps/CommunityApps/AddStorjNode.md:63]: Yes, it is for the Storj app to use. In Angelfish it was 20988, but that no longer works they need to use 28967 now. The other is just the web port for the app to use.

**Review by DjP-iX — APPROVED** [review]

**Pull request merged**